optuna/trial.py

```diff
@@ -362,8 +362,8 @@ def report(self, value, step=None):
         if step is not None:
             self.storage.set_trial_intermediate_value(self._trial_id, step, value)
 
-    def should_prune(self, step):
-        # type: (int) -> bool
+    def should_prune(self, step=None):
```

Comment:
Could you modify [BaseTrial.should_prune](https://github.com/pfnet/optuna/blob/568046a089f8f52fdbd66354d663d887a75880b5/optuna/trial.py#L59) method as well?